File: Solutions/1630. Arithmetic Subarrays/1630. Arithmetic Subarrays.java
class Solution {
    public List<Boolean> checkArithmeticSubarrays(int[] nums, int[] l, int[] r) {
        List<Boolean> res=new ArrayList<>();
        for(int i=0;i<l.length;++i){
            int st=l[i],ed=r[i];
            int[] temp=new int[ed-st+1];
            for(int s=0,k=st;k<=ed;++s,++k)
                temp[s]=nums[k];
            Arrays.sort(temp);
            int j=1;
            for(;j<temp.length;++j){
                if(temp[j]-temp[j-1]!=temp[1]-temp[0]){
                    res.add(false);
                    break;
                }
            }
            if(j==temp.length)
                res.add(true);
        }
        return res;
    }
}

File: Solutions/1630. Arithmetic Subarrays/1630. Arithmetic Subarrays.cpp
class Solution {
public:
    vector<bool> checkArithmeticSubarrays(vector<int>& nums, vector<int>& l, vector<int>& r) {
        vector<bool> res(l.size());
        for(int i=0;i<l.size();++i){
            vector<int> temp(begin(nums)+l[i],begin(nums)+r[i]+1);
            sort(begin(temp),end(temp));
            res[i]=true;
            for(int j=1;j<temp.size();++j){
                if(temp[j]-temp[j-1]!=temp[1]-temp[0]){
                    res[i]=false;
                    break;
                }
            }
        }
        return res;
    }
};